### DISCUSSION TITLE
Clearing cache on close

### DISCUSSION THREAD

**Discussion by almetria** [BrowserWorks/waterfox#4104] [Ideas]
I realize there's already an option to clear cookies and cache on program close. Having to log in to frequently visited websites every time I open the browser, most with 2FA, especially multiple times a day, doesn't appeal. It'd be lovely to have the option to just clear cache without clearing cookies on program close.

**Comment by IdrisQe** [comment]
I kind of asked for similar in part of my suggestion here: https://github.com/BrowserWorks/Waterfox/issues/3549
No responses, which I get since there's more important work being done, but I do hope something like this gets implemented, a way to keep specific preferences, data, and cookies, because yeah, I have the exact same issue. I don't like keeping my browser open nonstop, I don't want to keep all cookies and/or site data, and there are several websites I log into that require annoying verification methods if I don't keep their data...

There _is_ an exceptions menu for cookie deletion, but unfortunately that list of exceptions itself is cleared when you clear site data, making it kind of useless for our purpose.
